test/unit/lib/createCustomFetchSpec.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const test = require('ava');
+const http = require('http');
+const createCustomFetch = require('../../../lib/createCustomFetch');
+
+test('verify auth header', async (t) => {
+    const port = 3000;
+
+    const server = http.createServer((req, res) => {
+        res.end(req.headers.authorization);
+    });
+
+    server.listen(port);
+
+    const hasAuth = await new Promise(async (resolve) => {
+        const fetch = createCustomFetch({ username: 'test', password: 'test' });
+        const res = await fetch(`http://localhost:${port}`);
+        resolve(Boolean(await res.text()));
+    });
+
+    const hasNoAuth = await new Promise(async (resolve) => {
+        const fetch = createCustomFetch();
+        const res = await fetch(`http://localhost:${port}`);
+        resolve(Boolean(await res.text()));
+    });
+
+    const hasAuthWithHeader = await new Promise(async (resolve) => {
+        const fetch = createCustomFetch({ username: 'test', password: 'test' });
+        const res = await fetch(`http://localhost:${port}`, { headers: { 'x-test': 1 } });
+        resolve(Boolean(await res.text()));
+    });
+
+    const hasNoAuthWithHeader = await new Promise(async (resolve) => {
+        const fetch = createCustomFetch();
+        const res = await fetch(`http://localhost:${port}`, { headers: { 'x-test': 1 } });
+        resolve(Boolean(await res.text()));
+    });
```

Comment:
These functions look all quite similar, so I would suggest to use an ava test macro to avoid duplication.

For example:
```js
async function customFetchMacro(t, { customFetchOptions, headers, expectedResult }) {
     // start server
    const fetch = createCustomFetch(customFetchOptions);
    const response = await fetch(`http://localhost:${port}`, { headers });
    const result = Boolean(await res.text());

    t.is(result, expectedResult);
    // close server
};

test('has auth', customFetchMacro, {
    customFetchOptions: { username: 'test', password: 'test' },
    expectedResult: true
});

test('has no auth', customFetchMacro, {
    expectedResult: false
});

test('has auth with header', customFetchMacro, {
    customFetchOptions: { username: 'test', password: 'test' },
    headers: { 'x-test': 1  },
    expectedResult: true
});

test('has auth with header', customFetchMacro, {
    headers: { 'x-test': 1  },
    expectedResult: false
});
```